x-pack/solutions/observability/plugins/streams/server/lib/streams/ingest_pipelines/generate_ingest_pipeline.ts

```diff
@@ -5,17 +5,50 @@
  * 2.0.
  */
 
-import { StreamDefinition, isRoot } from '@kbn/streams-schema';
+import {
+  StreamDefinition,
+  getParentId,
+  isRoot,
+  isWiredStreamDefinition,
+} from '@kbn/streams-schema';
+import { IngestPutPipelineRequest } from '@elastic/elasticsearch/lib/api/types';
 import { ASSET_VERSION } from '../../../../common/constants';
 import { logsDefaultPipelineProcessors } from './logs_default_pipeline';
 import { getProcessingPipelineName } from './name';
 import { formatToIngestProcessors } from '../helpers/processing';
 
-export function generateIngestPipeline(id: string, definition: StreamDefinition) {
+export function generateIngestPipeline(
+  id: string,
+  definition: StreamDefinition
+): IngestPutPipelineRequest {
+  const isWiredStream = isWiredStreamDefinition(definition);
   return {
     id: getProcessingPipelineName(id),
     processors: [
       ...(isRoot(definition.name) ? logsDefaultPipelineProcessors : []),
+      ...(!isRoot(definition.name) && isWiredStream
+        ? [
+            {
+              script: {
+                source: `
```

Comment:
is it explicit in the design of streams that you cannot send data to non-root streams?

Reply:
(or route documents from one stream/data stream to a stream that isn't a direct child)?

Reply:
Yes, 4th item here: https://github.com/elastic/streams-program/blob/main/docs/architecture/README.md#wired-streams-invariants

Reply:
It's debatable whether we want to keep it like this, but for now it is a constraint and I rather drop a limitation later on than having to deal forever with too lose of a definition :)

Reply:
yeh, that makes sense. took me a minute to understand how the script worked, but it's quite nice really. i like it.

Reply:
TIL https://www.hyrumslaw.com/

Reply:
Always a great quote 😁